packages/react-native/src/Authenticator/common/FederatedProviderButton/types.ts

```diff
@@ -0,0 +1,19 @@
+import {
+  ImageSourcePropType,
+  ImageStyle,
+  StyleProp,
+  TextStyle,
+  ViewStyle,
+} from 'react-native';
+
+import { ButtonProps } from '../../../primitives';
+
+export interface FederatedProviderButtonProps extends ButtonProps {
+  source: ImageSourcePropType;
+}
+
+export interface FederatedProviderButtonStyles {
+  container: ViewStyle;
+  icon: StyleProp<ImageStyle>;
```

Comment:
Just curious, why do we need the `StyleProp` here?

Reply:
We could just do `ImageStyle`